include/deal.II/lac/ginkgo_solver.h

```diff
@@ -219,11 +225,294 @@ namespace GinkgoWrappers
      * parameters and setup the CG solver from the CG factory which solves the
      * linear system.
      *
-     * @p executor The execution paradigm for the CG solver.
+     * @p exec_type The execution paradigm for the CG solver.
+     */
+    SolverCG(SolverControl &       solver_control,
+             std::string           exec_type,
+             const AdditionalData &data = AdditionalData());
+
+    /**
+     * Constructor.
+     *
+     * @p solver_control The solver control object is then used to set the
+     * parameters and setup the CG solver from the CG factory which solves the
+     * linear system.
+     *
+     * @p exec_type The execution paradigm for the CG solver.
+     *
+     * @p preconditioner The preconditioner for the solver.
+     */
+    SolverCG(SolverControl &                    solver_control,
+             std::string                        exec_type,
+             std::shared_ptr<gko::LinOpFactory> preconditioner,
+             const AdditionalData &             data = AdditionalData());
+
+  protected:
+    /**
+     * Store a copy of the settings for this particular solver.
+     */
+    const AdditionalData additional_data;
+  };
+
+
+  /**
+   * An implementation of the solver interface using the Ginkgo BICGSTAB solver.
+   *
+   * @ingroup GinkgoWrappers
+   */
+  template <typename ValueType = double, typename IndexType = int32_t>
+  class SolverBICGSTAB : public SolverBase<ValueType, IndexType>
+  {
+  public:
+    /**
+     * A standardized data struct to pipe additional data to the solver.
+     */
+    struct AdditionalData
+    {};
+
+    /**
+     * Constructor.
+     *
+     * @p solver_control The solver control object is then used to set the
+     * parameters and setup the BICGSTAB solver from the BICGSTAB factory which
+     * solves the linear system.
+     *
+     * @p exec_type The execution paradigm for the BICGSTAB solver.
+     */
+    SolverBICGSTAB(SolverControl &       solver_control,
+                   std::string           exec_type,
+                   const AdditionalData &data = AdditionalData());
+
+    /**
+     * Constructor.
+     *
+     * @p solver_control The solver control object is then used to set the
+     * parameters and setup the BICGSTAB solver from the BICGSTAB factory which
+     * solves the linear system.
+     *
+     * @p exec_type The execution paradigm for the BICGSTAB solver.
+     *
+     * @p preconditioner The preconditioner for the solver.
+     */
+    SolverBICGSTAB(SolverControl &                    solver_control,
+                   std::string                        exec_type,
+                   std::shared_ptr<gko::LinOpFactory> preconditioner,
```

Comment:
same comments here in all regards. probably the same for the various classes below as well.

Reply:
It would also be good to have the class names align with those already used for linear solver's in deal.II. For example, we use `SolverBicgstab` for all of the BiCG (stabilised) solvers. Perhaps you could also check this for the others (I've not looked myself).

Reply:
Thanks. I did check for most solvers and thought that everything was capitalized, but I missed Bicgstab.